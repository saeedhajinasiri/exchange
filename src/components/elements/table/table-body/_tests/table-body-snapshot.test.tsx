import React from "react";
import {cleanup, render} from "test";
import TableBody from '../table-body'
import TableRow from "../../table-row";
import Table from "../../table";

afterEach(cleanup)

describe('Test TableBody UI', () => {
  it('Default', () => {
    const {asFragment} = render(<Table><TableBody/></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with className prop', () => {
    const {asFragment} = render(<Table><TableBody className={'test-class-name'}/></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with children prop', () => {
    const children = <TableRow/>;
    const {asFragment} = render(<Table><TableBody>{children}</TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test TableBody Colors', () => {
  const children = <TableRow color={"primary"}/>
  it('color=primary', () => {
    const {asFragment} = render(<Table><TableBody color={"primary"}>{children}</TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=secondary', () => {
    const {asFragment} = render(<Table><TableBody color={"secondary"}>{children}</TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=tertiary', () => {
    const {asFragment} = render(<Table><TableBody color={'tertiary'}>{children}</TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=grey', () => {
    const {asFragment} = render(<Table><TableBody color={'grey'}>{children}</TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=light', () => {
    const {asFragment} = render(<Table><TableBody color={'light'}>{children}</TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=dark', () => {
    const {asFragment} = render(<Table><TableBody color={'dark'}>{children}</TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });
});
