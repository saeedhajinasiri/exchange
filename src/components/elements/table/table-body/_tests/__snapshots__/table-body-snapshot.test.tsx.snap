// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TableBody Colors color=dark 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
      color="dark"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-primary border-primary"
      />
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableBody Colors color=grey 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
      color="grey"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-primary border-primary"
      />
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableBody Colors color=light 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
      color="light"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-primary border-primary"
      />
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableBody Colors color=primary 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
      color="primary"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-primary border-primary"
      />
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableBody Colors color=secondary 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
      color="secondary"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-primary border-primary"
      />
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableBody Colors color=tertiary 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
      color="tertiary"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-primary border-primary"
      />
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableBody UI Default 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    />
  </table>
</DocumentFragment>
`;

exports[`Test TableBody UI Test with children prop 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-white border-grey-light"
      />
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableBody UI Test with className prop 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group test-class-name"
    />
  </table>
</DocumentFragment>
`;
